design-proposals/storage-migration/storage-migration.md

```diff
@@ -0,0 +1,282 @@
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: dest-pvc
+spec:
+  accessModes:
+    - ReadWriteOnce
+  volumeMode: Filesystem
+  resources:
+    requests:
+      storage: 1Gi
+---
+apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+metadata:
+  annotations:
+    kubevirt.io/libvirt-log-filters: "1:qemu.qemu_monitor 1:*"
+  labels:
+    kubevirt.io/vm: vm-alpine-datavolume
+  name: vm-alpine-datavolume
+spec:
+  dataVolumeTemplates:
+  - metadata:
+      creationTimestamp: null
+      name: source-pvc
+    spec:
+      pvc:
+        accessModes:
+        - ReadWriteOnce
+        resources:
+          requests:
+            storage: 1Gi
+        storageClassName: local
+      source:
+        registry:
+          url: docker://registry:5000/kubevirt/alpine-container-disk-demo:devel
+  running: true
+  template:
+    metadata:
+      labels:
+        kubevirt.io/vm: vm-alpine-datavolume
+    spec:
+      domain:
+        devices:
+          disks:
+          - disk:
+              bus: virtio
+            name: datavolumedisk1
+          interfaces:
+          - masquerade: {}
+            name: default
+        resources:
+          requests:
+            memory: 128Mi
+      networks:
+      - name: default
+        pod: {}
+      terminationGracePeriodSeconds: 0
+      volumes:
+      - dataVolume:
+          name: source-pvc
+        name: datavolumedisk1
+
+```
+
+Trigger the migration with:
+```yaml
+apiVersion: storage.kubevirt.io/v1alpha1
+kind: StorageMigration
+metadata:
+  name: sc-mig
+spec:
+  vmiName: vmi
+  migrationVolumes:
+  - sourcePvc: source-pvc
+    destinationPvc: dest-pvc
+```
+
+Once the migration succeed:
+
+Changes on the VMI:
+```yaml
+
+apiVersion: v1
+items:
+- apiVersion: kubevirt.io/v1
+  kind: VirtualMachineInstance
+[...]
+    volumes:
+    - name: datavolumedisk1
+      persistentVolumeClaim:
+        claimName: pvc-dest
+ status:
+    migrationState:
+      completed: true
+      endTimestamp: "2023-08-17T08:14:45Z"
+ [...]
+      migrationVolumes:
+      - destinationPvc: pvc-dest
+        sourcePvc: source-pvc
+
+```
+
+On the VM:
+```yaml
+apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+[...]
+      volumes:
+      - name: datavolumedisk1
+        persistentVolumeClaim:
+          claimName: pvc-dest
+
+```
+
+## Scalability
+
+This proposal shouldn't have any impact on scalability.
+
+## Update/Rollback Compatibility
+
+Old KubeVirt versions won't be able to use this feature and the new added fields
+in the API shouldn't interfere with previous versions.
+
+## Functional Testing Approach
+
+Extensively add functional tests for migrating VMs with non-shared storage:
+- Migrating from a filesystem PVC to another filesystem PVC
+- Migrating from a block PVC to a block PVC
+- Migrating from a block PVC to a filesystem PVC
+- Migrating from a filesystem PVC to a block PVC
+- Migrating from a PVC to a larger PVC
+- Migrating from a filesystem storage class to another filesystem storage class
```

Comment:
Have to make sure to account for different filesystem overhead on source/target filesystems